skimage/measure/tests/test_fit.py

```diff
@@ -386,3 +386,11 @@ def residuals(self, data):
     data = np.arange(4)
     ransac(data, DummyModel, min_samples=3, residual_threshold=0.0,
            max_trials=10)
+
+
+def test_ransac_with_no_final_inliers():
+    data = np.random.rand(5, 2)
+    model, inliers = ransac(data, model_class=LineModelND, min_samples=3,
```

Comment:
One thing I don't understand here is that our tests should fail if a warning is not caught in a specific context (like for example https://github.com/scikit-image/scikit-image/blob/master/skimage/transform/tests/test_warps.py#L666). Are you sure that the warning is issued? Maybe I'm missing something here...

Reply:
hmm, I definitely see the warning I run the test

```
Testing started at 18:33 ...
/home/wwymak/anaconda3/envs/skimage-dev/bin/python /snap/pycharm-professional/206/plugins/python/helpers/pycharm/_jb_pytest_runner.py --target test_fit.py::test_ransac_with_no_final_inliers
Launching pytest with arguments test_fit.py::test_ransac_with_no_final_inliers in /home/wwymak/libraries/scikit-image/skimage/measure/tests

============================= test session starts ==============================
platform linux -- Python 3.7.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1 -- /home/wwymak/anaconda3/envs/skimage-dev/bin/python
cachedir: .pytest_cache
rootdir: /home/wwymak/libraries/scikit-image/skimage/measure
collecting ... collected 1 item

test_fit.py::test_ransac_with_no_final_inliers PASSED                    [100%]

=============================== warnings summary ===============================
tests/test_fit.py::test_ransac_with_no_final_inliers
  /home/wwymak/libraries/scikit-image/skimage/measure/fit.py:879: UserWarning: No inliers found. Model not fitted
    warn("No inliers found. Model not fitted")

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================= 1 passed, 1 warning in 0.19s =========================

Process finished with exit code 0
```
:thinking:

Reply:
The warning is issued indeed, but (apparently) this counts as passed.  I guess that's an upstream (not directly related) issue/choice regarding whether a warning should be considered a pass or a fail. In any case, it's cleaner to catch the warning in the dedicated context, so that we have a 'plain' pass.

Reply:
edit: "whether a warning ..." --> "which kind of warning exactly should be treated as an error"

Reply:
The line break suggested here was not made, but it is within the <80 character limit as is so I will mark this as resolved